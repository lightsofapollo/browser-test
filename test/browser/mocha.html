<!DOCTYPE HTML>
<html>
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
  <title>body</title>
</head>
<body>
<div id="mocha"></div>
<script src="/browser_test/browser_test.js" type="text/javascript" charset="utf-8"></script>
<script src="/node_modules/mocha/mocha.js" type="text/javascript" charset="utf-8"></script>
<script type="text/javascript" charset="utf-8">

function Reporter(runner) {
  runner.on('start', function(evt) {
    browserTest.createSuite(this.total, 'mocha.html');
  });

  runner.on('test', function(evt) {
    browserTest.createTest(evt.fullTitle());
  });

  runner.on('pass', function(evt) {
    browserTest.passTest();
  });

  runner.on('fail', function(evt, err) {
    browserTest.failTest(err);
  });

  runner.on('pending', function(evt) {
    browserTest.createTest(evt.title)
    browserTest.pendingTest();
  });

  runner.once('end', function() {
    browserTest.finishSuite();
  });
}

mocha.setup({
  ui: 'tdd',
  reporter: Reporter
});

suite('I have mochas', function() {
  test('woot', function() {
  });

  test.skip('pending', function() {
  });

  test('fail', function() {
    throw new Error('error')
  });

  test('async fail', function(done) {
    setTimeout(function() {
      throw new Error('async fail');
    });
  });
});

mocha.run();
</script>
</body>
</html>

