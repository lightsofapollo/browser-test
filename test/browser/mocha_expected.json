{
  "ok": false,
  "asserts": [
    {
      "ok": true,
      "number": 1,
      "name": "I have mochas woot"
    },
    {
      "ok": true,
      "number": 2,
      "name": "pending # SKIP"
    },
    {
      "ok": false,
      "number": 3,
      "name": "I have mochas fail"
    },
    {
      "ok": false,
      "number": 4,
      "name": "I have mochas async fail"
    }
  ],
  "pass": [
    {
      "ok": true,
      "number": 1,
      "name": "I have mochas woot"
    },
    {
      "ok": true,
      "number": 2,
      "name": "pending # SKIP"
    }
  ],
  "fail": [
    {
      "ok": false,
      "number": 3,
      "name": "I have mochas fail"
    },
    {
      "ok": false,
      "number": 4,
      "name": "I have mochas async fail"
    }
  ],
  "todo": [],
  "errors": [],
  "plan": {
    "start": 1,
    "end": 4
  }
}
